packages/mobx-state-tree/src/core/node/BaseNode.ts

```diff
@@ -6,40 +6,77 @@ import {
     Hook,
     escapeJsonPath
 } from "../../internal"
-import { createAtom } from "mobx"
+import { createAtom, IAtom } from "mobx"
 
 /**
  * @internal
  * @private
  */
 export abstract class BaseNode {
-    protected readonly subpathAtom = createAtom(`path`)
-    protected escapedSubpath: string
-    state = NodeLifeCycle.INITIALIZING
+    private _escapedSubpath?: string
+
+    private _subpath!: string
+    get subpath() {
+        return this._subpath
+    }
+
     storedValue: any
 
+    private aliveAtom?: IAtom
+    private _state = NodeLifeCycle.INITIALIZING
+    get state() {
+        return this._state
+    }
+    set state(val: NodeLifeCycle) {
+        const wasAlive = this.isAlive
+        this._state = val
+        const isAlive = this.isAlive
+
+        if (this.aliveAtom && wasAlive !== isAlive) {
+            this.aliveAtom.reportChanged()
+        }
+    }
+
     readonly type: IAnyType
     readonly hookSubscribers: { [k: string]: EventHandler<(node: any, hook: Hook) => void> } = {}
 
     environment: any = undefined
-    subpath: string = ""
-    parent: ObjectNode | null = null
+
+    private _parent!: ObjectNode | null
+    get parent() {
+        return this._parent
+    }
 
     constructor(type: IAnyType, parent: ObjectNode | null, subpath: string, environment: any) {
         this.environment = environment
         this.type = type
-        this.parent = parent
-        this.subpath = subpath
-        this.escapedSubpath = escapeJsonPath(this.subpath)
+        this.partialSetParent(parent, subpath)
+    }
+
+    private pathAtom?: IAtom
+    protected partialSetParent(parent: ObjectNode | null, subpath: string) {
```

Comment:
note the most descriptive name. What is "partial" about it? Probably deserves a clearer name or a comment :)

Reply:
That there's an abstract setParent, what's a good prefix for internal functions? An underscore? Internal? I'm open to suggestions

Reply:
I typically use base as well for such cases indeed

Reply:
sounds good, renamed partialXXX to baseXXX